lnbits/wallets/base.py

```diff
@@ -91,9 +97,19 @@ def get_payment_status(
     ) -> Coroutine[None, None, PaymentStatus]:
         pass
 
-    @abstractmethod
-    def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
-        pass
+    async def paid_invoices_stream(self) -> AsyncGenerator[str, None]:
+        while True:
+            for invoice in self.pending_invoices:
```

Comment:
this loop is non-blocking, the while loop will be CPU intensive

Reply:
Why would it be cpu intensive? There is a sleep at the bottom.

The original queue was never put into. A better overall approach would probably be creating a task for each invoice which only queries the invoice status for a certain time and then stops, similar to the frontend.

Reply:
> Why would it be cpu intensive? There is a sleep at the bottom.

My bad. I thought the sleep is in the catch block.